### PR TITLE
Add overview of features to readme

### PR DESCRIPTION
**Pull request by timeyyy**
This pr aims to open up dialogue about how best to market nim, the Why.

Ideally the overview and features presented should be chosen to match that story, The How

- [ ] Concise overview of the nim compiler
- [x] Point to more information
- [ ] Show the vision for nim
- [ ] Show features targeting that vision
- [ ] Move out information on nim development to wiki

### CODE REVIEW

**Comment by timeyyy** [comment]
I have pieced together some initial ideas based on the website and the wiki. If nim is indeed targeting the game industry compile speeds could be something worth talking about. Are compiler speeds being benchmarked and checked for regressions?

**Comment by Araq** [comment]
> Are compiler speeds being benchmarked and checked for regressions?

The bootstrapping process reports timings and memory usage for this reason and I keep track of these numbers. So the answer is "yes" but it could be automated more.

**Comment by stisa** [comment]
Nice work!  
Some suggestions:
- add a link to the [stdlib](http://nim-lang.org/docs/lib.html) near awesome-nim.
- I'd say the overview should only list high level features, maybe with a couple specific examples, btw of you keep it, I think 'high performance game' would sound better than 'high resource game'
- use h3 headers ( ### ) instead of bold in the features section.
- mention [ide support](https://github.com/nim-lang/Nim/wiki/editor-support) (maybe in tooling?)

**Comment by timeyyy** [comment]
Whats the main target area of nim? If we have that defined we can filter features to present.

**Comment by Araq** [comment]
> Whats the main target area of nim? If we have that defined we can filter features to present.

What is the main target area of C++? It's only been used for high performance computing, OSes, browers, game engines, Google's core algorithms, web crawlers, business applications ... 

"It's good for everything" is bad for marketing though, so let's say it tries to excel/excels at async IO and C++ interop.

**Comment by timeyyy** [comment]
I've never written c++, will this c [example](https://rosettacode.org/wiki/Call_a_foreign-language_function#Nim) also work for c++?

**Comment by Araq** [comment]
> I've never written c++, will this c example also work for c++?

Yes pretty much, but you have to use ``importcpp``.

**Comment by timeyyy** [comment]
> "It's good for everything" is bad for marketing though

Targetting a single area will also alow tooling and libraries to grow in one area meaning instead of doing 30% of what everyone wants we can do 110% of what 5% of people wants. This polish in one area means industry is more liable to start using nim.

Gamedev is a nice target specially if we can utilize all those c++ libraries. @araq mentioned another ui library a few days ago. Could be a viable market as there are alot of apps being made with webkit and electron(web technologies) producing binaries of 80mb. Kivy for python is the only alternative I found for developing apps quickly but it is implemented in python (sdl2) so it is not usable by other languages. A kivy lib
 clone in nim would be cool...

**Comment by Araq** [comment]
> Targetting a single area will also alow tooling and libraries to grow in one area meaning instead of doing 30% of what everyone wants we can do 110% of what 5% of people wants. This polish in one area means industry is more liable to start using nim.

Agreed.

File: readme.md [hunk 1]
```diff
@@ -1,9 +1,63 @@
...
+  - [Small Binary](https://hookrace.net/blog/nim-binary-size/)
+  - [Write Tracking](http://nim-lang.org/blog/writetracking.html) - immutability
+
+**Langauge**
+  - [Concurrent](https://rosettacode.org/wiki/Synchronous_concurrency)
+  - [Domain Specific Languages](http://andreaferretti.github.io/rosencrantz/)
+  - [Hygenic Macros](http://nim-lang.org/docs/macros.html)
+  - [Low level memory allocation](https://rosettacode.org/wiki/Memory_allocation#Nim)
+  - [Type inference](https://nim-by-example.github.io/variables/type_casting_inference/) - Strongly typed
+  - Link to more features with examples such as  https://haxe.org/documentation/introduction/language-features.html
```
Comment (andreaferretti) [inline readme.md]: Haxe? :-?

Comment (timeyyy) [inline readme.md]: it's a todo item. :)

File: readme.md [hunk 2]
```diff
@@ -1,9 +1,63 @@
 # <img src="https://raw.githubusercontent.com/nim-lang/assets/master/Art/logo-crown.png" width="36"> Nim [![Build Status](https://travis-ci.org/nim-lang/Nim.svg?branch=devel)](https://travis-ci.org/nim-lang/Nim)
+Efficient like C, expressive like Python and flexible like Lisp.
+
+```python
```
Comment (yglukhov) [inline readme.md]: s/python/nim

Comment (dom96) [inline readme.md]: I don't think code examples belong in the Github readme.

File: readme.md [hunk 3]
```diff
@@ -1,9 +1,63 @@
...
+```
+
+
+For more indepth information check [Nim's website](http://nim-lang.org).
+Other curated resources can be found at [Awesome Nim](https://github.com/VPashkov/awesome-nim)
+
+
+## Overview
+
+Nim is a modern general purpose programming language excelling in Async IO and C++ interop. These features make nim very suitable for:
```
Comment (dom96) [inline readme.md]: > Async IO and C++ interop.

I know @Araq said this but I find this lame.

Comment (Araq) [inline readme.md]: So say something better.

File: readme.md [hunk 4]
```diff
@@ -1,9 +1,63 @@
...
+  - [Editor Support](https://github.com/nim-lang/Nim/wiki/editor-support)
+  - [Suggest](https://github.com/nim-lang/nimsuggest)
+
+
+## What's being worked on Now
+  - link to highlevel roadmap, plan of what will be included, the vision!
+  - [Technical Roadmap](https://github.com/nim-lang/Nim/wiki/Roadmap)
+
+## Limitations
+  - Tail call optimization
```
Comment (yglukhov) [inline readme.md]: IMO this is not worth mentioning. Even though Nim itself doesn't do tail call optimization, it still can be done by underlying C compiler.

**Pull request closed without merging**